### DISCUSSION TITLE
Some logs not written on first time configurations

### DISCUSSION THREAD

**Discussion by ramtech123** [syslog-ng/syslog-ng#4241] [Q&A] [answered]
Hi Team,

I facing some weird issue with respect to getting logs written to some of the events. More details are shared at https://stackoverflow.com/questions/74763257/syslog-ng-logs-not-processing-certain-logs-possibly-due-to-journal-cursor-issue.

Summary: Certain logs from auth/authpriv facility are not written on first time configuration (either to the file or remote syslog servers). Mostly syslog-ng not even considering these events, but are visible in journalctl logs. Automated "syslog-ng-ctl reload" or "systemctl restart syslog-ng" do not resolve the issue. However, manual restart of syslog-ng makes things work, all the logs are recieved and processed by syslog-ng as expected.

Any help and pointers would be appreciated.

**Comment by bazsi** [comment, marked as answer]
This is probably caused by your real time clock going backwards. The notification mechanism in libsystemd does not work in this case.

There's a proof-of-concept patch in this syslog-ng issue:

https://github.com/syslog-ng/syslog-ng/issues/2836

But I've increased the priority to tackle that problem and fix this, as it is causing issue more often than I anticipated.

As a workaround you should synchronize the time for your VM, preferably so that during boot it waits until a sync and then keep the time synchronized by ntp.
